examples/seq2seq/finetune.py

```diff
@@ -209,18 +209,21 @@ def calc_generative_metrics(self, preds, target) -> Dict:
 
     def _generative_step(self, batch: dict) -> dict:
         t0 = time.time()
-        generated_ids = self.model.generate(
-            batch["input_ids"],
-            attention_mask=batch["attention_mask"],
-            use_cache=True,
-            decoder_start_token_id=self.decoder_start_token_id,
-            num_beams=self.eval_beams,
-        )
+        loss_tensors, logits = self._step(batch)
+        if self.eval_beams == 0:
```

Comment:
IMO would be better to add an explicit argument like `predict_with_generate` to decide weather to use generate or not, this will also align it with #6769. Setting `eval_beams` to 0 , will cause [this](https://github.com/huggingface/transformers/blob/master/examples/seq2seq/finetune.py#L116) assert to fail

Reply:
@patil-suraj : I updated that assertion above :).

Good point on creating an additional arg, though. I was veering on the side of fewer args, but this might also be something that's useful as an explicit arg than a hidden one. I don't have a strong opinion either way.

Reply:
aah right, I missed the updated assertion .

Right, not super important. Just a nit